synapse/crypto/context_factory.py

```diff
@@ -43,16 +56,18 @@ def configure_context(context, config):
         try:
             _ecCurve = crypto.get_elliptic_curve(_defaultCurveName)
             context.set_tmp_ecdh(_ecCurve)
-
         except Exception:
             logger.exception("Failed to enable elliptic curve for TLS")
-        context.set_options(SSL.OP_NO_SSLv2 | SSL.OP_NO_SSLv3)
+
+        context.set_options(
+            SSL.OP_NO_SSLv2 | SSL.OP_NO_SSLv3 | SSL.OP_NO_TLSv1 | SSL.OP_NO_TLSv1_1
```

Comment:
Why are we setting `NO_TLSv1`? I though the point was to optionally support it?

Reply:
This is server, not client. Our server cipher string requires TLS 1.2 anyway to negotiate any of the ciphers, so this just makes the handshake a little simpler.